src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,83 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Entry<K, V>[] table;
+    private int size;
+
+    private static class Entry<K, V> {
+        private final K key;
+        private V value;
+        private final Entry<K, V> next;
+
+        private Entry(K key, V value, Entry<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        this.table = new Entry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= LOAD_FACTOR * table.length) {
+            resize();
+        }
+        int index = hashIndex(key);
+        Entry<K, V> entry = table[index];
+        while (entry != null) {
+            if (Objects.equals(entry.key, key)) {
+                entry.value = value;
+                return;
+            }
+            entry = entry.next;
+        }
+        Entry<K, V> node = table[index];
+        add(index, key, value, node);
```

Comment:
```suggestion
        add(index, key, value);
```
You do not need it, cause it always be null here

Reply:
In this case there will be a compilation error